Library/Homebrew/brew.rb

```diff
@@ -28,7 +21,12 @@
 class MissingEnvironmentVariables < RuntimeError; end
 
 begin
+  require "stackprof" if ENV["HOMEBREW_STACKPROF"]
   require_relative "global"
```

Comment:
Ideally want to be able to profile this because it's really slow so would need `StackProf.start(mode: :wall, raw: true)` to be before this line.

Reply:
Would it be unreasonable to put the `require "stackprof"` and `StackProf.start(mode: :wall, raw: true)` lines in an `if ENV["HOMEBREW_STACKPROF"]` in [`startup.rb`](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/startup.rb)or [`load_path.rb`](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/standalone/load_path.rb)? That we we can profile _most_ of `require_relative "global"`. However, that might not be good enough because the [`Homebrew.setup_gem_environment!(setup_path: false)`](https://github.com/Homebrew/brew/blob/ca58ef59cb1b619f21f589edb93137e6d6898915/Library/Homebrew/standalone/load_path.rb#L9) won't be profiled.

@hyuraku sorry for the bad guidance but I'm thinking it may be better to go back to the way you were doing it before. Or, another option instead of using `ENV["HOMEBREW_LIBRARY"]` might be to just define `HOMEBREW_LIBRARY_PATH` is `brew.rb` (like is done [here](https://github.com/Homebrew/brew/blob/ca58ef59cb1b619f21f589edb93137e6d6898915/Library/Homebrew/standalone/load_path.rb#L9)). That way, no need to be finicky with env variables.